### PR TITLE
[processor/geoip] Move stability to alpha

### PR DESCRIPTION
**Pull request by rogercoll**
**Description:** <Describe what has changed.>

Moves its stability to alpha because the basic functionality is already covered (local geoIP database + context to where apply the transformations)

**Link to tracking Issue:** <Issue number if applicable> https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34737

**Testing:** <Describe what testing was performed and which tests were added.> NA

**Documentation:** <Describe the documentation added.> Updated using mdatagen

### CODE REVIEW

File: processor/geoipprocessor/metadata.yaml [hunk 1]
```diff
@@ -3,6 +3,6 @@ type: geoip
 status:
   class: processor
   stability:
-    development: [traces, metrics, logs]
+    alpha: [traces, metrics, logs]
```
Comment (crobert-1) [inline processor/geoipprocessor/metadata.yaml:6]: I believe we want to add `contrib` to the list of distributions now, and then open a PR against the releases repo as explained [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#releasing-new-components).

**Review by crobert-1 — COMMENTED** [review]
The processor needs to be added to https://github.com/open-telemetry/opentelemetry-collector-contrib/blame/main/reports/distributions/contrib.yaml as well. 👍

File: processor/geoipprocessor/metadata.yaml (hunk 1 above)
Comment (rogercoll) [reply processor/geoipprocessor/metadata.yaml:6]: Good catch! Added in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34788/commits/c9be10ffb294a5b4d8afccffb98337a0d214a697

Comment (rogercoll) [reply processor/geoipprocessor/metadata.yaml:6]: PR in the releases repository https://github.com/open-telemetry/opentelemetry-collector-releases/pull/646

**Review by andrzej-stencel — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**